src/anniversary/sf33rd/Source/Game/PulPul.c

```diff
@@ -1,190 +1,652 @@
 
-INCLUDE_ASM("asm/anniversary/nonmatchings/sf33rd/Source/Game/PulPul", init_pulpul_round2);
+    vib_req[ix][0] = vib_req[ix][1] = 0;
+
+    init_pulpul_round2(ix);
+}
+
+void init_pulpul_round2(s16 ix) {
+    ppwork[ix].vital = 0xA0;
+    pul[ix].ix.cal = 0;
+}
 
-#if defined(TARGET_PS2)
-INCLUDE_ASM("asm/anniversary/nonmatchings/sf33rd/Source/Game/PulPul", pulpul_stop);
-#else
 void pulpul_stop() {
-    not_implemented(__func__);
+    pulpul_stop2(0);
+    pulpul_stop2(1);
 }
-#endif
 
-#if defined(TARGET_PS2)
-INCLUDE_ASM("asm/anniversary/nonmatchings/sf33rd/Source/Game/PulPul", pulpul_stop2);
-#else
-void pulpul_stop2(s32 ix) {
-    not_implemented(__func__);
+void pulpul_stop2(s16 ix) {
+    if (ppwork[ix].ok_dev) {
+        vibParamTrans(ix, &pulpara[1]);
+        ppwork[ix].inStop = 4;
+    }
+    
+    ppwork[ix].ok_dev = 0;
+    ppwork[ix].p[0].ppnew = 0;
+
+    waitVsyncDummy();
 }
+
+void pulpul_request(s16 id, s16 ix) {
+#if defined(TARGET_PS2)
+    s32 pulpul_req_copy();
 #endif
+    PULREQ* adr;
 
-INCLUDE_ASM("asm/anniversary/nonmatchings/sf33rd/Source/Game/PulPul", pulpul_request);
+    if (vib_sel[id] == 0) {
+        return;
+    }
 
-INCLUDE_ASM("asm/anniversary/nonmatchings/sf33rd/Source/Game/PulPul", pulpul_req_copy);
+    if (ix == 0) {
+        return;
+    }
 
-#if defined(TARGET_PS2)
-INCLUDE_ASM("asm/anniversary/nonmatchings/sf33rd/Source/Game/PulPul", pp_vib_on);
-#else
-void pp_vib_on(s32 id) {
-    not_implemented(__func__);
+    if (test_flag) {
+        adr = &ot_pulreq[ix];
+        adr->adrs = (PPWORK_SUB_SUB*)&ot_pulreq_xx;
+    } else {
+        adr = &pulreq[ix];
+    }
+
+    vib_req[id][adr->prio] = ix;
+
+    pulpul_req_copy(id, adr);
 }
+
+void pulpul_req_copy(s16 id, PULREQ* adr) {
+    ppwork[id].p[adr->prio].ppnew = ppwork[id].ok_dev;
+    ppwork[id].p[adr->prio].rno[0] = 0;
+    ppwork[id].p[adr->prio].rno[1] = adr->rno;
+    ppwork[id].p[adr->prio].padr = adr->adrs;
+}
+
+void pp_vib_on(s16 id) {
+#if defined(TARGET_PS2)
+    s32 pulpul_req_copy();
 #endif
+    PULREQ* adr;
 
-INCLUDE_ASM("asm/anniversary/nonmatchings/sf33rd/Source/Game/PulPul", pulpul_request_again);
+    adr = &pulreq[3];
+    pulpul_req_copy(id, adr);
+}
 
+void pulpul_request_again() {
 #if defined(TARGET_PS2)
-INCLUDE_ASM("asm/anniversary/nonmatchings/sf33rd/Source/Game/PulPul", pp_operator_check_flag);
-#else
+    s32 pulpul_request();
+#endif
+    
+    pulpul_request(0, vib_req[0][0]);
+    pulpul_request(1, vib_req[1][0]);
+}
+
 void pp_operator_check_flag(u8 fl) {
-    not_implemented(__func__);
+    ppwork[0].opck = ppwork[1].opck = fl;
 }
-#endif
 
-#if defined(TARGET_PS2)
-INCLUDE_ASM("asm/anniversary/nonmatchings/sf33rd/Source/Game/PulPul", move_pulpul_work);
-#else
 void move_pulpul_work() {
-    not_implemented(__func__);
+    move_pulpul(ppwork);
+    move_pulpul(&ppwork[1]);
+
+    test_flag == 0;
 }
+
+s32 chkVibUnit(s32 port) {
+    if (((*flpad_adr)[port].kind == 0) || ((*flpad_adr)[port].kind) == 0x8000)
```

Comment:
```suggestion
    if ((flpad_adr[0][port].kind == 0) || (flpad_adr[0][port].kind) == 0x8000)
```

Same goes for other similar cases